x-pack/plugins/reporting/public/components/reporting_panel_content.tsx

```diff
@@ -40,24 +42,32 @@ class ReportingPanelContentUi extends Component<Props, State> {
 
     this.state = {
       isStale: false,
-      absoluteUrl: this.getAbsoluteReportGenerationUrl(props),
+      curlParams: this.getReportGenerationCurl(props),
       layoutId: '',
     };
   }
+  /*
+    curl --request POST 'http://u:p@localhost:5601/api/reporting/generate/printablePdf' \
+  --header 'Content-Type: application/json' \
+  --data-raw '{
+    "jobParams": "(browserTimezone:America/Los_Angeles,layout:(dimensions:(height:2024,width:1440),id:preserve_layout),objectType:dashboard,relativeUrls:!('\''/app/dashboards#/view/722b74f0-b882-11e8-a6d9-e546fe2bba5f?_g=(filters:!!(),refreshInterval:(pause:!!f,value:900000),time:(from:now-7d,to:now))&_a=(description:!'\''Analyze%20mock%20eCommerce%20orders%20and%20revenue!'\'',filters:!!(),fullScreenMode:!!f,options:(hidePanelTitles:!!f,useMargins:!!t),query:(language:kuery,query:!'\''!'\''),timeRestore:!!t,title:!'\''%5BeCommerce%5D%20Revenue%20Dashboard!'\'',viewMode:view)'\''),title:'\''[eCommerce] Revenue Dashboard'\'')"
+  }'
+  */
 
-  private getAbsoluteReportGenerationUrl = (props: Props) => {
-    const relativePath = this.props.apiClient.getReportingJobPath(
+  private getReportGenerationCurl = (props: Props) => {
+    const { url, payload } = this.props.apiClient.getReportingJobPath(
       props.reportType,
       props.getJobParams()
     );
-    return url.resolve(window.location.href, relativePath);
+
+    return { url, payload };
```

Comment:
I think the `payload` variable needs to be stringified or have escapes added. When I tried this out, I got a string that had unescaped single quotes, and it didn't work when I tried pasting it into the terminal.

Reply:
Yup, that's a big TODO for me. There's a mess of encoding issues since it's JSON embedded into a cURL call